File: Breadth-first_search(BFS)/python/02_바이러스.py
# fast campus 강의
# https://www.acmicpc.net/problem/2606
# 2

from collections import deque

n = int(input())
m = int(input())
adj = [[] for _ in range(n + 1)]

for i in range(m):
    start, end = map(int, input().split())
    adj[start].append(end)
    adj[end].append(start)

def bfs(v):
    global count
    q = deque([v])
    while q:
        v = q.popleft()
        if not(visited[v]):
            visited[v] = True
            count += 1
            for e in adj[v]:
                if not(visited[e]):
                    q.append(e)

visited = [False] * (n + 1)
count = 0
bfs(1)

print(count - 1)

File: Breadth-first_search(BFS)/c++/02_바이러스.cpp
// fast campus 강의
// https://www.acmicpc.net/problem/2606
// 2

#include <iostream>
#include <vector>
#include <queue>

using namespace std;

vector<int> virus[101];
bool visited[101] = { false, };
int N, M, result;

void bfs()
{
	queue<int> q;
	q.push(1);

	while (!q.empty()) {
		int x = q.front();
		q.pop();

		if (!visited[x]) {
			visited[x] = true;
			result++;

			for (auto next : virus[x])
				if (!visited[next])
					q.push(next);
		}
	}
}

int main()
{
	cin >> N >> M;

	for (int i = 0; i < M; i++) {
		int p, q;
		scanf("%d %d", &p, &q);

		virus[p].push_back(q);
		virus[q].push_back(p);
	}

	bfs();
	cout << result - 1;
}